Networking/Sources/Networking/Peer.swift

```diff
@@ -271,29 +277,44 @@ final class PeerImpl<Handler: StreamHandler>: Sendable {
         }
     }
 
-    // TODO: Add reconnection attempts & Apply exponential backoff delay
     func reconnect(to address: NetAddr, role: PeerRole) throws {
-        logger.debug("reconnecting", metadata: ["to address": "\(address)", "role": "\(role)"])
-        try connections.write { connections in
-            if connections.byAddr[address] != nil {
-                logger.warning("reconnecting to \(address) already connected")
-                return
+        let state = reconnectStates.read { reconnectStates in
+            reconnectStates[address] ?? ReconnectState(attempt: 0, delay: 1)
```

Comment:
```suggestion
            reconnectStates[address] ?? .init()
```

and have a `init` constructor to setup a default delay